GOVERNANCE.md

```diff
@@ -0,0 +1,221 @@
+Apart from following the Code of Conduct, there are no special requirements.
+As users continue to engage and become more involved with the Flux community, they may find themselves becoming contributors.
+
+### Contributors
+
+Flux welcomes all kinds of contributions, including code, issues, documentation, external tools, advocacy and community work.
+Contributors are accepted based on the merit of their contributions.
+Contributions help Maintainers ensure that the project continues to satisfy end user needs.
+Teams defined below may have different contribution processes and requirements for the various areas of Flux appropriate to their team.
+
+### Maintainers
+
+Maintainers are elected Contributors who serve on one or more Teams.
+For the election process, see Membership Changes below.
+
+Maintainers are expected to:
+
+- enable and promote Flux community Values
+- engage with end Users through appropriate communication channels appropriate to their teams
+- serve as a point of conflict resolution between Contributors to the Flux area appropriate to their teams
+- maintain open collaboration within and across teams
+- additional responsibilites are defined per team below
+
+## Teams
+
+The Flux project is governed by Maintainer teams, each with defined scope and resonsibilities.
+Teams are responsible for the overall organization, code development, community management, and security.
+
+The goal of governing by teams is to allow parts of the project to act independently as appropriate, enabling the project to grow horizontally while remaining well-maintained.
+
+Teams below are also mapped to github teams within the `fluxcd` github org.
+Their members are mapped to github user accounts.
+
+### Org team
+
+@fluxcd/org <https://github.com/orgs/fluxcd/teams/org>
+
+Scope and responsibilities:
+
+- shared Maintainers responsibilies, as well as
+- overseeing the project health and growth
+- maintaining the brand, mission, vision, values, and scope of the overall project
+- changes to licencing and intellectual property
+- administering access to all project assets
+- handling code of conduct violations
+- managing financial decisions
+- defining the scope and responsibilities of all other teams
+- making descisions that affect multiple teams
+- resolving escalated team decisions when the teams responsible are blocked
+- changes to governance (this document)
+
+### Dev team
+
+@fluxcd/dev <https://github.com/orgs/fluxcd/teams/dev>
+
+Scope and responsibilities:
+
+- shared Maintainers responsibilies, as well as
+- all technical decisions related to project software
+- administering, creating, archiving, and deleting source code repositories
+- release management
+
+In future, the scope and responsibilities of this team may be split into separate dev-related teams.
+But even if so, due to Flux componentized architecture, there should be no assumption of a 1:1 relationship between a source repository and a team.
```

Comment:
Does this translate to: "one may be a member of the dev team, but only a mainainer of a single Flux project (e.g. `fluxcd/X-controller`, but no others)"?

Reply:
Ah, no. What does was intended to convey is that even if in future it becomes necessary for the Dev team to split into multiple projects￼ (example: toolkit team, flagger team, helm controller team...), that a git repo does not necessarily equal a "team". For example, a toolkit team would oversee several git repos as part of its scope (fluxcd/toolkit, fluxcd/go-git-providers, etc).

If this feels too clunky we can clarify in some way or just remove it. ￼I added this to preempt confusion because a number of CNCF projects governance assumes a one to one relationship between a git repo and a team or sub-project, but over time or in reality that's not always the case. In the case of Flux, we already know it's not.

Reply:
Makes sense now that I have read it again, together with your explanation.

What I am missing, given the current state of how Maintainers are already divided across project and how GitHub works with granting permissions to teams, is a section covering "outside maintainers" (as GitHub calls them): people that only maintain a single repository, but do not have rights across the full `fluxcd` GitHub org.

Mainly, because adding them to the Dev team would grant them permissions to:

> - administering, creating, archiving, and deleting source code repositories

which are rights I think we do not want to hand out generously.

Options that come to mind to solve this:
- Already introducing "teams" or sub-projects in the GOVERNANCE doc that have rights scoped to the set of repositories they maintain
- Cover "outside maintainers" in the document, and explain that they are leveled between contributors and the Dev team.
- Make them part of the GitHub org (which gives recognition to people for their efforts, as they can publicly publish it on their profile), but do not assign them to a team and only give them access to the repositories they maintain. This would elevate them from "outside maintainer", while granting them scoped permissions.

Reply:
Yes I think Hidde is right. https://github.com/fluxcd/community/pull/13/files#diff-9e8744ec73212c1966bb19fa22fc3e0aR66 says Maintainers have to serve on one or more teams. I don't think we want to insist on this -- we want some repositories (I am thinking of go-git-providers) to be self-contained and self-sufficient.

Reply:
OK I have made small changes to address this:
- Moved admin/creating/archiving/deleting git repo responsibility to the Org team
- Clarified a Dev team member may be scoped to one or more git repos, defined by a MAINTAINERS file in each repo (as we do now). Also noted how this is decided (by existing Dev team members listed in a repo's MAINTAINERS file)

WDYT?